components/accelerator/nvidia/infiniband/component.go

```diff
@@ -58,7 +59,20 @@ func (c *component) States(ctx context.Context) ([]components.State, error) {
 }
 
 func (c *component) getStates(ctx context.Context, thresholds infiniband.ExpectedPortStates) ([]components.State, error) {
-	o, err := infiniband.GetIbstatOutput(ctx, []string{c.toolOverwrites.IbstatCommand})
+	var o *infiniband.IbstatOutput
+	var err error
+	for i := 0; i < 3; i++ {
+		o, err = infiniband.GetIbstatOutput(ctx, []string{c.toolOverwrites.IbstatCommand})
+		if err == nil || !errors.Is(err, infiniband.ErrIbstatOutputEmpty) {
+			break
+		}
+		log.Logger.Warnw("ibstat output is empty, retrying", "attempt", i+1, "error", err)
```

Comment:
It doesn't make sense unless we can figure out the root cause of why we got empty output. If we add a retry here, similar scenarios should follow the same logic, but that would be too strange.

Reply:
removed